lib/rule-tester/rule-tester.js

```diff
@@ -305,6 +305,36 @@ function getCommentsDeprecation() {
     );
 }
 
+/**
+ * Emit a deprecation warning if function-style format is being used.
+ * @param {string} ruleName Name of the rule.
+ * @returns {void}
+ */
+function emitLegacyRuleAPIWarning(ruleName) {
+    if (!emitLegacyRuleAPIWarning[`warned-${ruleName}`]) {
+        emitLegacyRuleAPIWarning[`warned-${ruleName}`] = true;
+        process.emitWarning(
+            `"${ruleName}" rule is using the deprecated function-style format and will stop working in ESLint v9. Please use object-style format: https://eslint.org/docs/developer-guide/working-with-rules`,
+            "DeprecationWarning"
+        );
+    }
+}
+
+/**
+ * Emit a deprecation warning if rule has options but is missing the "meta.schema" property
+ * @param {string} ruleName Name of the rule.
+ * @returns {void}
+ */
+function emitMissingSchemaWarning(ruleName) {
```

Comment:
We can refactor, we will just have to create a unique flag every time. If we set `emitWarning[`warned-${ruleName}`] = true` in the helper it may suppress another warning coming from the same rule.

I suppose this can be the helper"

```js
// API
function emitWarning(flag, message) {
    if (!emitWarning[`warned-${flag}`]) {
        emitWarning[`warned-${flag}`] = true;
        process.emitWarning(
            `"${ruleName}" rule has options but is missing the "meta.schema" property and will stop working in ESLint v9. Please add a schema: https://eslint.org/docs/developer-guide/working-with-rules#options-schemas`,
            "DeprecationWarning"
        );
    }
}

// Usage
emitWarning(`legacy-rule-api-${ruleName}`, 'some message...');
emitWarning(`missing-schema-${ruleName}`, 'some message...');
```

Reply:
I feel we can add this helper later when we have requirements for adding warnings in other files too. 
/cc @mdjermanovic WDYT?

Reply:
> I feel we can add this helper later when we have requirements for adding warnings in other files too. /cc @mdjermanovic WDYT?

I agree. We could consider refactoring when we add more warnings.